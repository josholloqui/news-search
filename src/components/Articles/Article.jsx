import React from 'react';
import PropTypes from 'prop-types';

const Article = ({ title, author, description }) => (
  <>
    <h3>{title}</h3>
    <cite>By: {author}</cite>
    <p>Description: {description}</p>
  </>
);

Article.propTypes = {
  title: PropTypes.string.isRequired,
  author: PropTypes.string.isRequired,
  description: PropTypes.string.isRequired
};

export default Article;
