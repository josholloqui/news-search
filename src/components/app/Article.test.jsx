import React from 'react';
import { render, cleanup } from '@testing-library/react';
import Article from './Article';

describe('Article component', () => {
  afterEach(() => cleanup());
  it('renders Article', () => {
    const { asFragment } = render(
      <Article
        title="calvin"
        author="and"
        description="hobbs"
      />
    );

    expect(asFragment()).toMatchSnapshot();
  });
});
