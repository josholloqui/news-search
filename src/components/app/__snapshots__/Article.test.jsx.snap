// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Article component renders Article 1`] = `
<DocumentFragment>
  <h3>
    calvin
  </h3>
  <cite>
    By: and
  </cite>
  <p>
    Description: hobbs
  </p>
</DocumentFragment>
`;
